packages/storybook/src/executors/utils.ts

```diff
@@ -216,6 +224,58 @@ export function resolveCommonStorybookOptionMapper(
 
       buildProjectName = defaultProjectName;
 
+      if (defaultProjectName) {
+        logger.warn(
+          `
+          No projectBuildConfig was provided. Nx will use the defaultProject as set in your workspace configuration
+          to build Storybook. This can lead to compilation errors for Storybook.
+          
+          If you want to change this, you can run the command by providing the projectBuildConfig flag as follows:
+                 
+          nx ${context.targetName ? context.targetName : 'storybook'} ${
+            context.projectName
+          } --projectBuildConfig=${context.projectName}${
+            !buildable && storybookBuildTarget ? ':build-storybook' : ''
```

Comment:
I think we need to fix this. If you have this Storybook config:

```
"images": {
        "my-storybook": {
          "builder": "@nrwl/storybook:storybook",
          "options": {
            "uiFramework": "@storybook/angular",
            "port": 4400,
            "config": {
              "configFolder": "libs/images/.storybook"
            }
          },
          "configurations": {
            "ci": {
              "quiet": true
            }
          }
        },
        "my-build-storybook": {
          "builder": "@nrwl/storybook:build",
          "outputs": ["{options.outputPath}"],
          "options": {
            "uiFramework": "@storybook/angular",
            "outputPath": "dist/storybook/images",
            "config": {
              "configFolder": "libs/images/.storybook"
            }
          },
          "configurations": {
            "ci": {
              "quiet": true
            }
          }
        }
}
```

You would get a log like `nx my-storybook images --projectBuildConfig=images` which wouldn't actually work because the `images` project is not buildable/publishable.

So we should either log out in a more generic way that they should point to the corresponding Storybook build target, or improve the search for the Storybook serve & build target within the project targets. I added some comments related to that further down in the migrator. I think some logic there could be reused.